### ISSUE TITLE
Trying rule missing_space_before_subcommand taking so long

### ISSUE THREAD

**Issue by auspic7** [nvbn/thefuck#1120]
The output of `thefuck --version` (something like `The Fuck 3.1 using Python
3.5.0 and Bash 4.4.12(1)-release`):

    The Fuck 3.29 using Python 3.8.2 and ZSH 5.8

Your system (Debian 7, ArchLinux, Windows, etc.):

    ubuntu 20.04 on wsl2

How to reproduce the bug:

    env THEFUCK_DEBUG=true thefuck test

The output of The Fuck with `THEFUCK_DEBUG=true` exported (typically execute `export THEFUCK_DEBUG=true` in your shell before The Fuck):

    DEBUG: Trying rule: missing_space_before_subcommand; took: 0:00:08.341279
    No fucks given

Anything else you think is relevant:

I have no idea why this taking so long. anyone else having this problem?

**Comment by divykj** [comment]
This rule is using `get_all_executables` in which it scans all the directories in `PATH` for possible executables AFAIK, normally it shouldn't be a problem, but in WSL the path of your host Windows OS gets concatenated with the Ubuntu's path. So it has to scan more paths for executable (to find the correct one), hence it takes longer.

Could you try out running and timing a non-existent command, and check if that takes a long time to execute too?
Something like:
```shell
$ time abcd
```

**Comment by auspic7** [comment]
``` bash
$ time abcd 
zsh: command not found: abcd
abcd  0.00s user 0.01s system 18% cpu 0.049 total
```

non-existent command won't take a long time.

by the way, `env THEFUCK_DEBUG=true thefuck abcd` will still take longer time since Trying rule: no_command takes a lot.
```bash
DEBUG: Trying rule: no_command; took: 0:00:08.681820
```

**Comment by divykj** [comment]
I was not able to find a proper reason as to is why this happening. I think `thefuck` is just unoptimized at many places which are aggregating to make it slow.

I think, let's try a cProfile
```shell
python -m cProfile $(which thefuck) zzzzzz
```

**Comment by pdavies** [comment]
I think this might be a dupe of #1036. On WSL, usually `no_command` runs slowly for me, but I'm sure while testing that, once or twice I saw `missing_space_before_subcommand` running slowly instead. At a guess, memoization of `get_all_executables` means that whoever hits it first pays the performance penalty, and occasionally `missing_space_before_subcommand` gets there first.

The fix, as with #1036 currently, is to set `excluded_search_path_prefixes`. In my testing, `/mnt/c/WINDOWS` (and subdirectories) was overwhelmingly the culprit. Adding that has resulted in a massive speedup.

**Comment by scorphus** [comment]
Duplicate of #1036
